dbt/adapters/databricks/api_client.py

```diff
@@ -356,6 +361,93 @@ def cancel(self, run_id: str) -> None:
         if response.status_code != 200:
             raise DbtRuntimeError(f"Cancel run {run_id} failed.\n {response.content!r}")
 
+    def list_active_runs_for_job(self, job_id):
+        request_body = {
+            'job_id': job_id,
+            'active_only': True,
+        }
+        active_runs_response = self.session.get("/list", json=request_body)
+        if active_runs_response.status_code != 200:
+            raise DbtRuntimeError(f"Error getting active runs.\n {active_runs_response.content!r}")
+
+        return active_runs_response.json().get("runs", [])
+
+class JobPermissionsApi(DatabricksApi):
+    def __init__(self, session: Session, host: str):
+        super().__init__(session, host, "/api/2.0/permissions/jobs")
+
+    def put(self, job_id, access_control_list):
+        request_body = {
+            "access_control_list": access_control_list
+        }
+
+        response = self.session.put(f"/{job_id}", json=request_body)
+        logger.info(f"Workflow permissions update response={response.json()}")
+
+        if response.status_code != 200:
+            raise DbtRuntimeError(f"Error updating Databricks workflow.\n {response.content!r}")
+
+    def get(self, job_id):
+        response = self.session.get(f"/{job_id}")
+
+        if response.status_code != 200:
+            raise DbtRuntimeError(f"Error fetching Databricks workflow permissions.\n {response.content!r}")
+
+        return response.json()
+
+
+class WorkflowJobApi(DatabricksApi):
+
+    def __init__(self, session: Session, host: str):
+        super().__init__(session, host, "/api/2.1/jobs")
+
+    def search_by_name(self, job_name: str) -> [Dict[str, Any]]:
+        response = self.session.get("/list", json={"name": job_name})
+
+        if response.status_code != 200:
+            raise DbtRuntimeError(f"Error fetching job by name.\n {response.content!r}")
+
+        logger.info(f"Job list response={response.json()}")
+        return response.json().get("jobs", [])
+
+    def create(self, job_spec: Dict[str, Any]) -> str:
+        """
+        :return: the job_id
+        """
+        response = self.session.post("/create", json=job_spec)
+
+        if response.status_code != 200:
+            raise DbtRuntimeError(f"Error creating Workflow.\n {response.content!r}")
+
+        logger.info(f"Workflow creation response={response.json()}")
+        return response.json()["job_id"]
+
+    def update_by_reset(self, job_id, job_spec):
```

Comment:
What is this intended for?  It's obscure enough that a doc string might be useful.

Reply:
Yeah good call. It's basically a PUT, but it's called a reset in the Databricks API. It's to make the job reflect the config, with any changes.

Reply:
Changed method name to `update_job_settings` per [your other comment](https://github.com/databricks/dbt-databricks/pull/762#discussion_r1783546514)